react/core/main.tsx

```diff
@@ -155,6 +155,12 @@ function validateRootComponent(rootName: string, extensions: Extensions) {
   }
 }
 
+function setLazyCookie(setCookie: string) {
+  if (setCookie !== null) {
+    document.cookie = setCookie
```

Comment:
Is this really setting the raw value of the cookies? I thought the browser would have some API for accessing the cookies already parsed only.

Also, here you are *setting* the whole cookies, while I believe we should only *append* the lazy cookies to the existing ones. In that case, it'd probably be necessary to check if the cookie we are setting is already set.

Reply:
Just for the last part, I think that this is not necessary as router won't set the header `X-Vtex-Lazy-Cookie` if it checks the existence of the cookie - we do the "cookie-stuff" inside `FindWorkspaceFromCookies`, where we first check if we can get the Workspace from cookies and then immediately returns the value.

Reply:
I don't know if I understood everything bothering you but I think that is everything ok. Just take a look here 
https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie

Reading it I concluded that the cookie is being set correctly and we don't have to actually care about the other ones that were set before (even after).

Reply:
Yep. @victorges go back to your golang cave 😘 hahahahaha 😬 😂